tests/api_tests.py

```diff
@@ -32,37 +33,65 @@ def setUp(self):
             'password': 'secret',
             'confirm_password': 'secret'
         }
-        save_user = user();
+        save_user = user()
         save_user.save({
-            'names':self.sample_user['names'],
-            'email':self.sample_user['email'],
-            'password':self.sample_user['password'],
-            'confirm_password':self.sample_user['confirm_password']
+            'id': self.sample_user['id'],
+            'names': self.sample_user['names'],
+            'email': self.sample_user['email'],
+            'password': self.sample_user['password'],
+            'confirm_password': self.sample_user['confirm_password']
         })
 
     def test_registration(self):
         '''
             Testing registration
         '''
-        response = self.app.post(self.url_prefix+'auth/register',data = {
-            'names':self.exist_user['names'],
-            'email':self.exist_user['email'],
-            'password':self.exist_user['password'],
-            'confirm_password':self.exist_user['confirm_password']
+        # Test registration with complete data
+        response = self.app.post(self.url_prefix + 'auth/register', data={
+            'names': self.exist_user['names'],
+            'email': self.exist_user['email'],
+            'password': self.exist_user['password'],
+            'confirm_password': self.exist_user['confirm_password']
         })
         self.assertEqual(response.status_code, 200)
         self.assertIn(b'have been successfully registered', response.data)
+        # Test registration with incomplete data
+        incomplete_data_response = self.app.post(self.url_prefix + 'auth/register', data={
+            'names': 'dummy name',
+            'confirm_password': self.exist_user['confirm_password']
+        })
+        self.assertEqual(incomplete_data_response.status_code, 400)
+        self.assertIn(b'Please provide', incomplete_data_response.data)
 
```

Comment:
What if an already registered user tries to register with the same credentials? Ensure all edge cases are covered in your tests

Reply:
Fixed